sdk/communication/azure-communication-identity/azure/communication/identity/_shared/utils.py

```diff
@@ -67,6 +68,20 @@ def get_current_utc_as_int():
     return _convert_datetime_to_utc_int(current_utc_datetime)
 
 
+def verify_datetime_format(input_datetime):
```

Comment:
This change seems unrelated to the stated purpose of the PR to reduce repeated code. I'd move it to a separate PR to have one feature per PR and keep focus on it

Reply:
we're tracking a separate issue for this https://github.com/Azure/azure-sdk-for-python/issues/28034

the change here is required in order for the pipelines to pass. we might exclude the change if it gets fixed in the other issue first.